### PR TITLE
Migrates manageFeatures to permissionsClass

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

This PR migrates `manageFeatures` to the `PermissionsClass` and in doing so, introduces 4 new helper methods: `canViewFeatureModal`, `canCreateFeature`, `canDeleteFeature`, `canUpdateFeature`. This level of granularity will be helpful as we introduce more fine-grain permissions.

### Dependencies

~~This PR is dependent on [PR-2292](https://github.com/growthbook/growthbook/pull/2292) being merged first, given the changes that PR makes to `Layout.tsx` and `SidebarLink.tsx`.~~

### Testing

- See test suite

### CODE REVIEW

File: packages/back-end/src/controllers/features.ts
Comment (mknowlton89) [inline packages/back-end/src/controllers/features.ts:511]: We are using this pattern a lot in this file - where we're checking if they have permission to update the feature, but it's changing an element of the feature that isn't something that if changed, would change their permission level. Open to ideas if this pattern is confusing to others.

File: packages/back-end/src/api/features/updateFeature.ts
Comment (jdorn) [inline packages/back-end/src/api/features/updateFeature.ts:38]: This is removing the checks above for `publishFeatures`.  Add those back in

File: packages/front-end/components/Layout/Layout.tsx
Comment (jdorn) [inline packages/front-end/components/Layout/Layout.tsx:215]: The importer doesn't use the current project at all.  It creates new projects based on the ones in LaunchDarkly and adds features there.  So we really do need to check for global permissions here.

**Review by jdorn — APPROVED** [review]
Added 2 small comments to fix.  Approving now so you don't need to wait for a re-review when you fix them.

**Pull request merged**
